### DISCUSSION TITLE
Option for using actual_data in validation for Reduced Models

### DISCUSSION THREAD

**Discussion by ilkersigirci** [alan-turing-institute/sktime#2938] [Q&A]
I would like to do multistep forecasting for reduced sklearn models. In the validation part (backtest, evaluate method), since we know the real `y` values, I would expect the model to use it instead of its predicted value for multistep forecast. Currently, reduced models recursively forecast one step for loop of `fh` in their `_predict` function.

For example, assume `fh=12`. And our data is X_train=[100,101,102] X_test=[103,104] .

In validation part, we train the model with `X_train` and we forecast the first value(103) and put store it. And for the second forecast point(104), we append X_test `103` point to X_train and make the new forecast accordingly without retrain the model. 

At first, I thought this can be implemented with ExpandingWindowsSplitter but this way, we retrain the model in every step.

```python
from sklearn.ensemble import RandomForestRegressor
from sktime.forecasting.model_evaluation import evaluate
from sktime.forecasting.model_selection import ExpandingWindowSplitter
from sktime.utils.plotting import plot_series
from sktime.datasets import load_airline
import numpy as np
from sktime.forecasting.compose import (
    make_reduction
)
y = load_airline()

forecaster = make_reduction(
            estimator=RandomForestRegressor(),
            scitype="tabular-regressor",
            window_length=1,
            strategy="recursive",
        )

cv = ExpandingWindowSplitter(
    step_length=1, fh=np.arange(1, 13), initial_window=72
)

df = evaluate(forecaster=forecaster, y=y, cv=cv, strategy="refit", return_data=True)

df.head(10)
```

I want to note that the current approach of sktime is absolutely true for real forecasting, where we don't know the future values of `y`. But in the validation part since we do know real value, I think it would be better to use them when calculating validation score and hence based on this score, selecting parameter with GridSearch etc.

What do you think about my proposed approach? This is a bad practice for backtesting?

Related sources:
- [1](https://machinelearningmastery.com/backtest-machine-learning-models-time-series-forecasting/)
- [2](https://towardsdatascience.com/proper-validation-of-a-time-series-model-5c1b54f43e60)
- [3](https://robjhyndman.com/hyndsight/tscv/)
- [4](https://otexts.com/fpp3/tscv.html)

**Comment by fkiraly** [comment]
hm, what you are suggesting would amount to not updating the reduction forecaster on the test period, is that correct?

In-principle, this can be achieved by setting `update_params=False` in the `update` function. If I understand correctly what you want to do for the reduction forecaster?

You can't use `evaluate` for that, since it does not allow you to control the `update_params` argument, but that could be achieved with a very simple modification to `evaluate` which controls the `update_params` flag in `update`, e.g., another strategy `"no-update"` which sets `update_params=False`.

**Comment by ilkersigirci** [comment]
It is correct. In my local environment, I am already using the similar way  that use suggested `update_params=False`. It would be good to add this functionality to sktime, though. Moreover, I would like to hear everyone's thought about this backtesting strategy. Whether using actual value for validation is good strategy?

**Reply by fkiraly** [reply]
Here are two PR with options, @ilkersigirci:
* a wrapper that turns off updating. This way you can compare forecasters that update with those that don't, subject to the same interface: #2955
* the `no-update` option for `evaluate` #2954

**Reply by fkiraly** [reply]
>Whether using actual value for validation is good strategy?

I guess that depends - what do you want to evaluate, or what is the "deployment situation" that you want an estimate of performance for? If you turn off update, it will give you a statement about what happens if you don't update your forecasters in deployment either, no?
